### ISSUE TITLE
Upgrade stuck

### ISSUE THREAD

**Issue by faust64** [openshift/elasticsearch-operator#203]
As of today, I can see to ElasticSearch operators listed in my ClusterServiceVersions:

```
# oc get csv
NAME                                        DISPLAY                        VERSION              REPLACES                              PHASE
elasticsearch-operator.4.2.1-201910221723   Elasticsearch Operator         4.2.1-201910221723                                         Succeeded
elasticsearch-operator.4.2.4-201911050122   Elasticsearch Operator         4.2.4-201911050122                                         Pending
```

Describing that pending CSV, I would find in its status:

```
  - lastTransitionTime: "2019-11-21T15:18:04Z"
    lastUpdateTime: "2019-11-21T15:18:04Z"
    message: conflicting CRD owner in namespace
    phase: Failed
    reason: OwnerConflict
  - lastTransitionTime: "2019-11-21T15:18:07Z"
    lastUpdateTime: "2019-11-21T15:18:07Z"
    message: 'installing: ComponentMissing: missing deployment with name=elasticsearch-operator'
    phase: Pending
    reason: NeedsReinstall
```

I tried to label the elasticsearch-operator deployment, in the openshift-operators namespace, with name=elasticseach-operator. Did not help.

Looking at the logs from the catalog-operator, in openshift-operator-lifecycle-manager project, we would see:

```
E1121 15:12:21.581706       1 queueinformer_operator.go:282] sync {"update" "openshift-operators"} failed: logging.openshift.io/v1/Elasticsearch (elasticsearches) already provided by elasticsearch-operator.4.2.4-201911050122
E1121 15:12:21.782068       1 queueinformer_operator.go:282] sync "openshift-operators" failed: logging.openshift.io/v1/Elasticsearch (elasticsearches) already provided by elasticsearch-operator.4.2.4-201911050122
E1121 15:12:23.381951       1 queueinformer_operator.go:282] sync {"update" "openshift-operators"} failed: logging.openshift.io/v1/Elasticsearch (elasticsearches) already provided by elasticsearch-operator.4.2.1-201910221723
E1121 15:12:23.781774       1 queueinformer_operator.go:282] sync "openshift-operators" failed: logging.openshift.io/v1/Elasticsearch (elasticsearches) already provided by elasticsearch-operator.4.2.1-201910221723
```

Having had a similar issue with Jaeger lately, I suspect this could be fixed by deinstalling then re-installing the operator. Though I'ld rather try and figure out what went wrong.

Any clue what could be going on?

**Comment by pavolloffay** [comment]
The same issue reported here: https://issues.redhat.com/browse/MAISTRA-1044

cc @jcantrill

**Comment by jcantrill** [comment]
@pavolloffay this is the first I have seen this issue.  Could it be that something is declaring the elasticsearch-operator as a dependency and dragging it in?  Or maybe there is an error in OLM allowing it to be installed as namespaced operator?

**Comment by faust64** [comment]
Unclear. Lately, a newer version was successfully applied (4.2.11-201912100122).

Some Pending copies were still showing (4.2.4-201911050122 ). So I deleted those ClusterServiceVersions (in several namespaces, not all). Right now, I no longer see this.

To clarify, elasticsearch is not installed as a namespace operator, a single copy is running in the openshift-operators-redhat namespace. Although I do have one CSV per namespace.

Currently, I'm only using ElasticSearch alongside the Logging operator.
I've also seen that issue on some customer cluster, with Istio/no Logging.

**Comment by jcantrill** [comment]
@ecordell Are you able to comment regarding this issue?  One item to note is the failure is reporting in the `openshift-operators` namespace.  The install location has been changed to `openshift-operators-redhat` to allow scraping of metrics for RH supported operators.  I don't recall when we advised the switch but maybe we are in a place between deployed namespaces.

**Comment by humanistical** [comment]
I'm seeing this same issue upgrading from OS 4.2.18 -> 4.3.1. Not resolved after upgrade to 4.3.5. 

**Operator Event View**
```
ClusterServiceVersion elasticsearch-operator.4.3.5-202003020549
Namespace openshift-operators
7 minutes ago
Generated from operator-lifecycle-manager
16132 times in the last 7 hours
installing: ComponentMissing: missing deployment with name=elasticsearch-operator

ClusterServiceVersion elasticsearch-operator.4.3.5-202003020549
Namespace openshift-operators
2 minutes ago
Generated from operator-lifecycle-manager
16306 times in the last 7 hours
conflicting CRD owner in namespace
```

------
**Installed Operator View**
```
Elasticsearch Operator4.3.1-202002032140 provided by Red Hat, Inc | Namespace openshift-operators | Deployment elasticsearch-operator | Cannot update Catalog source was removed 
-- | -- | -- | -- | --
Elasticsearch Operator4.3.5-202003020549 provided by Red Hat, Inc | Namespace openshift-operators | Deployment elasticsearch-operator | Failed Up to date
-- | -- | -- | -- | --
```

**Comment by humanistical** [comment]
Resolved it by uninstalling from the OperatorHub, manually removing the old version from the openshift-operators namespace, then reinstalling the new one from OperatorHub. This cleaned it up across all namespaces
